dot/sync/chain_sync.go

```diff
@@ -166,26 +175,45 @@ func newChainSync(cfg chainSyncConfig) *chainSync {
 		workerPool:         newSyncWorkerPool(cfg.net, cfg.requestMaker),
 		badBlocks:          cfg.badBlocks,
 		requestMaker:       cfg.requestMaker,
+		waitPeersDuration:  cfg.waitPeersDuration,
 	}
 }
 
+func (cs *chainSync) waitEnoughPeersAndTarget() <-chan struct{} {
+	resultCh := make(chan struct{})
+
+	go func() {
+		defer close(resultCh)
+		for {
+			cs.workerPool.useConnectedPeers()
+			_, err := cs.getTarget()
+
+			totalAvailable := cs.workerPool.totalWorkers()
+			if totalAvailable >= uint(cs.minPeers) && err == nil {
+				return
+			}
+
+			time.Sleep(cs.waitPeersDuration)
+		}
+	}()
+
+	return resultCh
+}
+
 func (cs *chainSync) start() {
 	// since the default status from sync mode is syncMode(tip)
 	isSyncedGauge.Set(1)
 
+	cs.wg.Add(1)
+	go cs.pendingBlocks.run(cs.finalisedCh, cs.stopCh, &cs.wg)
+
 	// wait until we have a minimal workers in the sync worker pool
-	for {
-		totalAvailable := cs.workerPool.totalWorkers()
-		if totalAvailable >= uint(cs.minPeers) {
-			break
-		}
+	waitEnoughPeersC := cs.waitEnoughPeersAndTarget()
 
-		// TODO: https://github.com/ChainSafe/gossamer/issues/3402
-		time.Sleep(time.Second)
+	select {
+	case <-waitEnoughPeersC:
```

Comment:
I think you can revise `cs.waitEnoughPeersAndTarget()` to be a blocking function that also listens on the `cs.stopCh` to exit early.